imagebuildah/build.go

```diff
@@ -482,6 +482,13 @@ func (s *StageExecutor) volumeCacheRestore() error {
 // imagebuilder tells us the instruction was "ADD" and not "COPY".
 func (s *StageExecutor) Copy(excludes []string, copies ...imagebuilder.Copy) error {
 	for _, copy := range copies {
+		// If the destination is a symlink, convert it to a hardpath
+		// otherwise it will be overwritten.
+		if symLink, err := resolveSymlink(s.mountPoint, copy.Dest); err == nil {
+			copy.Dest = symLink
+		} else {
+			return errors.Wrapf(err, "error reading symbolic link to %q", copy.Dest)
+		}
```

Comment:
This is having troubles with a clause like
 COPY Dockerfile1.scratch /Dockerfile1
Rather than copying Dockerfile1.scratch to the file /Dockerfile1, it's creating a directory named /Dockerfile1. Will dive further on Monday.

Reply:
I think this now works when the dest is a symlink with this os.Stat check.